api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -264,4 +296,301 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() throws Exception {
+		assertEquals(20, conceptClasses.size());
+		assertEquals(conceptService.getConceptClass(20), conceptClasses.get(19));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#purgeConceptClass(ConceptClass)
+	 * @Verifies should purge concept class
+	 */
+	@Test
+	public void purgeConceptClass_shouldPurgeTheGivenConceptClass() throws Exception {
+		ConceptClass cc = conceptService.getConceptClass(1);
+		assertNotNull(cc);
+		
+		conceptService.purgeConceptClass(cc);
+		assertNull(conceptService.getConceptClass(1));
+		
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllConceptDatatypes()
+	 * @Verifies should return all concept datatypes
+	 */
+	@Test
+	public void getAllConceptDatatypes_shouldReturnAllConceptDataypes() throws Exception {
+		List<ConceptDatatype> conceptDatatypes = conceptService.getAllConceptDatatypes(true);
+		assertEquals(12, conceptDatatypes.size());
+		assertEquals(conceptService.getConceptDatatypeByUuid("8d4a4488-c2cc-11de-8d13-0010c6dffd0f"), conceptDatatypes.get(0));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getSetsContainingConcept(Concept)
+	 * @Verifies should return sets containing concept if given a valid concept
+	 */
+	@Test
+	public void getSetsContainingConcept_shouldReturnSetsIfGivenValidConcept() throws Exception {
+		Concept concept = conceptService.getConceptByUuid("0dde1358-7fcf-4341-a330-f119241a46e8");
+		List<ConceptSet> conceptSets = conceptService.getSetsContainingConcept(concept);
+		
+		assertNotNull(conceptSets);
+		assertEquals(conceptSets.get(0).getConcept(), concept);
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getSetsContainingConcept(Concept)
+	 * @Verifies should return empty list if no set contains the given concept
+	 */
+	@Test
+	public void getSetsContainingConcept_shouldReturnEmptyListIfGivenConceptWithNoSet() throws Exception {
+		Concept concept = conceptService.getConceptByUuid("0cbe2ed3-cd5f-4f46-9459-26127c9265ab");
+		List<ConceptSet> conceptSets = conceptService.getSetsContainingConcept(concept);
+		
+		assertEquals(conceptSets, java.util.Collections.emptyList());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getConcepts(String, Locale, boolean)
+	 * @Verifies return concepts in given locale
+	 */
+	@Test
+	public void getConcepts_shouldReturnConceptsInGivenLocale() throws Exception {
+		Locale locale = new Locale("en","GB");
+		List<Locale> locales = new Vector<Locale>();
+		locales.add(locale);
+		List<ConceptSearchResult> res = conceptService.getConcepts("CD4 COUNT", locales, true, null, null, null, null, null, null, null);
+		
+		assertEquals("CD4 COUNT",res.get(0).getConceptName().getName());
+		
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrigByIngredients(Concept)
+	 * @Verifies should fail if no ingredient is given
+	 */
+	@Test
+	public void getDrugsByIngredient_shouldFailIfGivenNoIngredient() throws Exception {
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("ingredient is required");
+		Concept nullConcept = null;
```

Comment:
If I use inline null here the test fails with hibernate.TransientObjectException as the null object is an unsaved transient instance